plugin-scalebar/src/main/java/com/mapbox/maps/plugin/scalebar/ScaleBarImpl.kt

```diff
@@ -312,20 +301,53 @@ class ScaleBarImpl : ScaleBar, View {
     canvas.drawText(text, x, y, textPaint)
   }
 
+  /**
+   * @return A tuple with the scale bar segment distance and the amount of bar segments
+   */
+  @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+  internal fun findSuitableScaleBarSegments(maxDistance: Float): Pair<Float, Int> {
+    var pair: Pair<Int, Int> = scaleTable[0]
+    for (i in 1 until scaleTable.size) {
```

Comment:
how often we call this method? Could there be performance issues with this code? maybe cache somehow index from scaleTable? or refactor it and store range for getting value with O(1)

Reply:
This method is called on each draw. Unless `ScaleBar.useContinuousRendering` is set to true the view is invalidated per need basis.

AFAIK there hasn't been issues with this piece of code (see previous implementation at [ScaleBarImpl#onDraw](https://github.com/mapbox/mapbox-maps-android/blob/63535428955bf5d4a968ddbe7f3de145e3b8e795/plugin-scalebar/src/main/java/com/mapbox/maps/plugin/scalebar/ScaleBarImpl.kt#L235) before.
I think that before we do any optimization we should measure it. I would leave it out of this PR.